pkg/controller/service_lb.go

```diff
@@ -0,0 +1,392 @@
+		klog.Errorf("failed to init route table, err: %v", err)
+		return err
+	}
+
+	providerName := getAttachNetworkProvider(svc)
+	attchIpAnnotation := fmt.Sprintf(util.IpAddressAnnotationTemplate, providerName)
+	attchCidrAnnotation := fmt.Sprintf(util.CidrAnnotationTemplate, providerName)
+	attchGatewayAnnotation := fmt.Sprintf(util.GatewayAnnotationTemplate, providerName)
+
+	if pod.Annotations[attchCidrAnnotation] == "" || pod.Annotations[attchGatewayAnnotation] == "" {
+		return fmt.Errorf("failed to get attachment network info for pod %s", pod.Name)
+	}
+
+	loadBalancerIP := pod.Annotations[attchIpAnnotation]
+	ipAddr := util.GetIpAddrWithMask(loadBalancerIP, pod.Annotations[attchCidrAnnotation])
+
+	var addRules []string
+	addRules = append(addRules, fmt.Sprintf("%s,%s", ipAddr, pod.Annotations[attchGatewayAnnotation]))
+	klog.Infof("add eip rules for lb svc pod, %v", addRules)
+	if err := c.execNatRules(pod, POD_EIP_ADD, addRules); err != nil {
+		klog.Errorf("failed to add eip for pod, err: %v", err)
+		return err
+	}
+
+	defaultGateway := pod.Annotations[util.GatewayAnnotation]
+	for _, port := range svc.Spec.Ports {
+		var protocol string
+		switch port.Protocol {
+		case corev1.ProtocolTCP:
+			protocol = "tcp"
+		case corev1.ProtocolUDP:
+			protocol = "udp"
+		case corev1.ProtocolSCTP:
+			protocol = "sctp"
+		}
+
+		var rules []string
+		rules = append(rules, fmt.Sprintf("%s,%d,%s,%s,%d,%s", loadBalancerIP, port.Port, protocol, svc.Spec.ClusterIP, port.TargetPort.IntVal, defaultGateway))
+		klog.Infof("add dnat rules for lb svc pod, %v", rules)
+		if err := c.execNatRules(pod, POD_DNAT_ADD, rules); err != nil {
+			klog.Errorf("failed to add dnat for pod, err: %v", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Controller) runUpdateLbSvcPodWorker() {
+	for c.processNextUpdateLbSvcPodWorkItem() {
+	}
+}
+
+func (c *Controller) processNextUpdateLbSvcPodWorkItem() bool {
+	obj, shutdown := c.updateLbSvcPodQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	err := func(obj interface{}) error {
+		defer c.updateLbSvcPodQueue.Done(obj)
+		var svc *corev1.Service
+		var ok bool
+		if svc, ok = obj.(*corev1.Service); !ok {
+			c.updateLbSvcPodQueue.Forget(obj)
+			utilruntime.HandleError(fmt.Errorf("expected Service in workqueue but got %#v", obj))
+			return nil
+		}
+
+		if err := c.handleUpdateLbSvcPod(svc); err != nil {
+			c.updateLbSvcPodQueue.AddRateLimited(obj)
+			return fmt.Errorf("error syncing '%s': %s, requeuing", svc.Name, err.Error())
+		}
+		c.updateLbSvcPodQueue.Forget(obj)
+		return nil
+	}(obj)
+
+	if err != nil {
+		utilruntime.HandleError(err)
+		return true
+	}
+	return true
+}
+
+func (c *Controller) handleUpdateLbSvcPod(svc *corev1.Service) error {
+	pod, err := c.getLbSvcPod(svc.Name, svc.Namespace)
+	if err != nil {
+		// There's no other way to delete update pod process in queue,so record err and return if pod does not exist
+		klog.Errorf("failed to get lb svc pod, %v", err)
+		return nil
```

Comment:
should requeue here

Reply:
modified. 
The new queue has been deleted and processed in create svc function.